webkitgtk.yml

```diff
@@ -0,0 +1,181 @@
+# And now, WebkitGTK and its deps *specifically.*
+# Big thanks to Telegram Desktop Webview's manifest maintainers for this.
+# Because I took pretty much the whole stupid thing and refactored it into its
+# own yml file. I then added many extra parts from all over the Flathub org.
+# Anyone who needs to include webkitgtk, *please* feel free to use this.
+
+name: webkitgtk
+buildsystem: cmake-ninja
+config-opts:
+  - -DCMAKE_BUILD_TYPE=RelWithDebInfo
+  - -DPORT=GTK
+  - -DUSE_SOUP2=ON  # This makes it build webkit2gtk-4.0 instead of 4.1.
+  - -DUSE_LIBSECRET=OFF
+  - -DUSE_WOFF2=OFF
+  - -DENABLE_GAMEPAD=OFF
+  - -DENABLE_INTROSPECTION=OFF
+  - -DENABLE_SPELLCHECK=OFF
+  - -DENABLE_DOCUMENTATION=OFF
+  - -DENABLE_MINIBROWSER=OFF
+sources:
+  - type: archive
+    url: https://webkitgtk.org/releases/webkitgtk-2.42.1.tar.xz
+    sha256: 6f41fac9989d3ee51c08c48de1d439cdeddecbc757e34b6180987d99b16d2499
+    x-checker-data:
+      type: html
+      url: https://webkitgtk.org/releases/
+      version-pattern: LATEST-STABLE-(\d[\.\d]+\d)
+      url-template: https://webkitgtk.org/releases/webkitgtk-$version.tar.xz
+  - type: shell
+    commands:
+      - sed -i 's@NAMES avif.h@NAMES avif/avif.h@' Source/cmake/FindAVIF.cmake
+      - sed -i '/PATH_SUFFIXES avif/d' Source/cmake/FindAVIF.cmake
+      - sed -i 's@${AVIF_INCLUDE_DIR}/avif.h@${AVIF_INCLUDE_DIR}/avif/avif.h@'
+        Source/cmake/FindAVIF.cmake
+modules:
+
+  # Geany exposes the host file system, so WebkitGTK **NEEDS** to be sandboxed.
+  - name: bubblewrap
+    buildsystem: meson
+    sources:
+      - type: archive
+        url: https://github.com/containers/bubblewrap/releases/download/v0.8.0/bubblewrap-0.8.0.tar.xz
+        sha256: 957ad1149db9033db88e988b12bcebe349a445e1efc8a9b59ad2939a113d333a
+        x-checker-data:
+          type: json
+          url: https://api.github.com/repos/containers/bubblewrap/releases/latest
+          tag-query: .tag_name
+          timestamp-query: .published_at
+          version-query: $tag | sub("^v"; "")
+          url-query: '"https://github.com/containers/bubblewrap/releases/download/\($tag)/bubblewrap-\($version).tar.xz"'
+  - name: xdg-dbus-proxy
+    buildsystem: meson
+    sources:
+      - type: archive
+        url: https://github.com/flatpak/xdg-dbus-proxy/releases/download/0.1.4/xdg-dbus-proxy-0.1.4.tar.xz
+        sha256: 1ec0eab53d1e49966d722352bcfd51ac402dce5190baedc749a8541e761670ab
+        x-checker-data:
+          type: json
+          url: https://api.github.com/repos/flatpak/xdg-dbus-proxy/releases/latest
+          tag-query: .tag_name
+          timestamp-query: .published_at
+          version-query: $tag
+          url-query: '"https://github.com/flatpak/xdg-dbus-proxy/releases/download/\($tag)/xdg-dbus-proxy-\($version).tar.xz"'
```

Comment:
This actually does nothing, you can delete these deps and build webkit with `-DENABLE_BUBBLEWRAP=OFF`. WebKit uses `flatpak-spawn` inside of flatpak.

Reply:
https://github.com/flathub/org.geany.Geany/pull/55

Reply:
Oh, you don't say? That makes sense. Thanks!